### ISSUE TITLE
Possibility to bypass image decoding [Feature] 

### ISSUE THREAD

**Issue by RuiAAPeres** [kean/Nuke#46]
As discussed @kean, ideally, one would be able to flag a `ImageRequest` to not be decoded. So after downloading the image, the corresponding blob of `NSData` would be kept inside the `NSURLCache`, but the decoding and posterior caching inside the `NSCache` would be bypassed.

**Comment by kean** [comment]
Here's a little more context about this issue: the problem is the lack of control over image preheating. Specifically, in some cases user would want preheating tasks to stop after image data is stored into disk cache.

**Comment by kean** [comment]
There is also a problem with Nuke design. It doesn't provide a straightforward way to implement this feature by extending existing classes. Adding this feature would require users to reimplement the entire `ImageLoader`. This problem should be addressed in Nuke v2.

**Comment by kean** [comment]
Closed by aca5899e6222fe72973a5873f59bfd3c94678f4f

**Comment by kean** [comment]
We came up with a simpler solution. Now there is a new property `memoryCacheStorageAllowed` in `ImageRequest` struct.

Usage:
```swift
// MARK: ImagePreheatingControllerDelegate
    
func preheatingController(controller: ImagePreheatingController, didUpdateWithAddedIndexPaths addedIndexPaths: [NSIndexPath], removedIndexPaths: [NSIndexPath]) {
    func requestForIndexPaths(indexPaths: [NSIndexPath]) -> [ImageRequest] {
        return indexPaths.map {
            var request = self.imageRequestWithURL(self.photos[$0.row])
            request.memoryCacheStorageAllowed = false
            request.shouldDecompressImage = false
            return request
        }
    }
    Nuke.startPreheatingImages(requestForIndexPaths(addedIndexPaths))
    Nuke.stopPreheatingImages(requestForIndexPaths(removedIndexPaths))
}
```

What happens: image data is loaded, stored in disk cache, decoded (NSData -> UIImage, neglectable) and NOT decompressed and NOT stored in memory.
